### PR TITLE
feat: Clearer colors for segmented button

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

I find the segmented button super hard to parse, especially when I was using a few of them next to each other.

Dark mode in particular is basically unusable

## Changes

* Swaps to use the primary state for the selected button
* Removes the non-3000 css
* Uses an overlap for the button with z-index to ensure the selected button is highlighted

|Before|After|
|----|----|
| Screenshot 2024-02-09 at 13 02 12 | Screenshot 2024-02-09 at 13 03 23 |
| Screenshot 2024-02-09 at 13 01 58 | Screenshot 2024-02-09 at 13 03 30 |

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

**Review by corywatilo — APPROVED** [review]
Getting rid of that mess of code feels so good!

**Review by daibhin — APPROVED** [review]
